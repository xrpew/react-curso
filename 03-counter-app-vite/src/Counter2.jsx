import React, { useState } from "react";

export const Counter2 = () => {
  const [ count, setCount ] = useState(0);

  const sum = () => setCount( count + 1 );

  return (
    <>
      <h1>{ count }</h1>
      <button onClick={ sum }> sumar </button>
    </>
  );
};
