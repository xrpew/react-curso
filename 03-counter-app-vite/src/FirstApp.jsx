import PropTypes from 'prop-types';


// const newMessage = {
//   message: "hello",
//   title: "sergio",
// };

const heloo = (a, b) => a + b;

// para declarar funciones que se usarán en el componente, hay que declararlas fuera de la función
export const FirstApp = ( {title, subtitle, name} ) => {


  return (
    <>
      <h1 data-testid='test-title'>{ title }</h1>
      {/* <p>{heloo('hola: ', 'bien y tu')}</p> */}
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      <p>{ subtitle }</p>
      <p>{ subtitle }</p>

      <p>{ name }</p>
    </>
  );
}; 


FirstApp.propTypes={

  title: PropTypes.string.isRequired,
  subTitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
  subTitle:8,
  name: 'sergio Perez',
  // title:'no hay título'
}