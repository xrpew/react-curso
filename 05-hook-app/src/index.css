body{
  padding: 20px;
}
button{
  margin-top: 10px;
  margin: 1vh;
}
a{
  margin-right: 5px;
}