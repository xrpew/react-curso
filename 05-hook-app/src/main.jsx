import React from "react";
import ReactDOM from "react-dom/client";
import { BrowserRouter } from 'react-router-dom'



import "./index.css";
// import { Memorize } from "./06-memo/Memorize";
// import { FocusScreen } from "./04-useRef/FocusScreen";
// import { Layout } from "./05-useLayoutEfect/Layout";
// import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks";
// import { FormWithCustomHooks } from "./02-useEffect/FormWithCustomHooks";
// import { CounterApp } from './01-useState/CounterApp'
// import { HooksApp } from './HooksApp'
// import { CounterWhitCustomHooks } from './01-useState/CounterWhitCustomHooks'
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { MemoHook } from "./06-memo/MemoHook";
// import { CallBackHook } from "./06-memo/CallBackHook";
// import './08-useReducer/intro-reducer'
// import { Padre } from "./07-tarea-memo/Padre";
// import { TodoApp } from "./08-useReducer/TodoApp";


import { MainApp } from "./09-useContext/MainApp";

ReactDOM.createRoot(document.getElementById("root")).render(

    <BrowserRouter>
        <MainApp/>
    </BrowserRouter>

);
