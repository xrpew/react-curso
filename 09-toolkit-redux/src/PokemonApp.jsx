import { useEffect } from "react"
import { useDispatch, useSelector } from "react-redux"
import { pokemonApi } from "./API/pokemonApi";
import { getPokemons } from "./store/slices/pokemon/thunks";

export const PokemonApp = () => {

  const distpach = useDispatch();
const { isLoading, pokemons = [] , page} = useSelector(state=> state.pokemons)

  useEffect(() => {
    distpach( getPokemons() );
    
  }, [])
  

  return (
    <>
    <h1>Pokemon APP</h1>
    <hr />
    <span> { isLoading && 'Cargando...'}</span>
    <ul>
      {pokemons.map(({name})=>(
        <li key={name}> {name} </li>
        ))
      }

    </ul>
    <button
        disabled={ isLoading }
        onClick={()=>distpach(getPokemons(page))}>
        
      Next
    </button>
    

    </>
    )
}
