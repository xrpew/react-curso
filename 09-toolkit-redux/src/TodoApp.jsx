import { useState } from "react";
import { useGetTodoQuery, useGetTodosQuery } from "./store/apis/todosApi"

export const TodoApp = () => {

    const [todoId, setTodoId] = useState(1) 
    const {isLoading, data: todo } = useGetTodoQuery( todoId );

    const nextTodo = ()=>{
        setTodoId(todoId + 1 )
    }
    const previousTodo = ()=>{
        setTodoId(todoId - 1 )
    }

  return (
        <>
            <h1>Todos ----- RTK Queri</h1>
            <hr />
            <h4>isLoading: {isLoading ? 'true' : 'false' }</h4>

            <pre>{JSON.stringify( todo )}</pre>

            <button
                onClick={previousTodo}>
                previous Todo
            </button>

            <button onClick={nextTodo}>
                next Todo
            </button>
            {/* <ul>
                {todos.map(todo=>(
                    <li key={todo.id}>
                        <strong>{todo.completed ? 'Dome' : 'Pending'}</strong>
                        <br />{todo.title} 
                    </li>

                ) )}

            </ul> */}


        
        </>

    )
}
