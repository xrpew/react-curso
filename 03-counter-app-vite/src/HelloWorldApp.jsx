import { FirstApp } from "./FirstApp"


export const HelloWorldApp = () => {
  return (
    <>
    <h1>Hello World App</h1>
    </>
  ) 
}
