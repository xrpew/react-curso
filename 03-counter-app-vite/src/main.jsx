import React from "react";
import ReactDOM from "react-dom/client";
import { Counter2 } from "./Counter2";
import { CounterApp } from "./CounterApp";
import { FirstApp } from "./FirstApp";
import { HelloWorldApp } from "./HelloWorldApp";
import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
     <FirstApp  title='hola' subTitle={5} name='sergio'/>
     <CounterApp value={0}/>
  </React.StrictMode>
);
 