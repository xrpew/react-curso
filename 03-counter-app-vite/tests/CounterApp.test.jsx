import { render, screen, fireEvent } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('probando el <CounterApp />',()=>{
    const Initialvalue = 10
    test('debe hacer match con el snapshot',()=>{
        const { container}= render(<CounterApp value={Initialvalue}/>)
        expect(container).toMatchSnapshot();
    });
    test('debe mostrar el valor inicial de 100',()=>{
        render(<CounterApp value={100}/>)
        expect( screen.getByText('100')).toBeTruthy()

    });
    test('debe de incrementar con el botón +1',()=>{
        render(<CounterApp value={ Initialvalue } /> );
        fireEvent.click( screen.getByText('+1') )  
        expect( screen.getByText('11')).toBeTruthy() 
    });
    test('debe de decrementar con el botón -1',()=>{
        render(<CounterApp value={ Initialvalue } /> );
        fireEvent.click( screen.getByText('-1') )  
        expect( screen.getByText('9')).toBeTruthy() 
    });
    test('debe de funcionar botón Reset',()=>{
        render(<CounterApp value={ Initialvalue } /> );
        fireEvent.click( screen.getByText('+1') ) 
        fireEvent.click( screen.getByText('+1') )  
        fireEvent.click( screen.getByText('+1') )  
        fireEvent.click( screen.getByText('Reset') )  

        expect( screen.getByText(Initialvalue)).toBeTruthy() 
    });
})