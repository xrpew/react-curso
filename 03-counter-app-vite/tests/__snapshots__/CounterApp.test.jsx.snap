// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando el <CounterApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button>
     -1 
  </button>
  <button>
     +1 
  </button>
  <button>
    Reset
  </button>
</div>
`;
